lightning/src/ln/channel.rs

```diff
@@ -4419,16 +4497,26 @@ impl<Signer: Sign> Channel<Signer> {
 				Some(InboundHTLCState::AwaitingAnnouncedRemoteRevoke(forward_info.clone()))
 			} else { None };
 			if let Some(state) = new_state {
+				log_trace!(logger, " ...promoting inbound AwaitingRemoteRevokeToAnnounce {} to AwaitingAnnouncedRemoteRevoke", log_bytes!(htlc.payment_hash.0));
 				htlc.state = state;
 			}
 		}
 		for htlc in self.pending_outbound_htlcs.iter_mut() {
 			if let Some(fail_reason) = if let &mut OutboundHTLCState::AwaitingRemoteRevokeToRemove(ref mut fail_reason) = &mut htlc.state {
 				Some(fail_reason.take())
 			} else { None } {
+				log_trace!(logger, " ...promoting outbound AwaitingRemoteRevokeToRemove {} to AwaitingRemovedRemoteRevoke", log_bytes!(htlc.payment_hash.0));
 				htlc.state = OutboundHTLCState::AwaitingRemovedRemoteRevoke(fail_reason);
 			}
 		}
+		if let Some((feerate, update_state)) = self.pending_update_fee {
+			if update_state == FeeUpdateState::AwaitingRemoteRevokeToAnnounce {
+				debug_assert!(!self.is_outbound());
+				log_trace!(logger, " ...promoting inbound AwaitingRemoteRevokeToAnnounce fee update {} to Committed", feerate);
+				self.feerate_per_kw = feerate;
+				self.pending_update_fee = None;
```

Comment:
Because `AwaitingRemoteRevokeToAnnounce` is handled differently in `build_commitment_transaction` - if we are rebuilding a previous commitment, we'll need to consider the state to decide whether to include it.